src/utils/extensions.ts

```diff
@@ -1,9 +1,42 @@
-import { AnyExtensionData, InstalledExtensionData } from '~types';
+import {
+  AnyExtensionData,
+  ExtensionConfig,
+  InstalledExtensionData,
+  ColonyExtension,
+  InstallableExtensionData,
+} from '~types';
 
 /**
- * Type guard to distinguish installed extension data from extension config
+ * Type guard to distinguish installed extension data from installable extension data
  */
 export const isInstalledExtensionData = (
   extension: AnyExtensionData,
 ): extension is InstalledExtensionData =>
   (extension as InstalledExtensionData).address !== undefined;
+
+export const mapToInstallableExtensionData = (
```

Comment:
For me, these functions make it less obvious what's happening in the hooks they're used in than if I were to see just the:

```
return [
          ...availableExtensions,
          ...extensionConfig,
          availableVersion: version,
        ];
```

Especially since it includes "map" in the title, which makes me think of a `.map` utility.

There is some code duplication within and between the `useExtension(s)Data` hooks, which may present a better opportunity to improve code readability. Or did you have something else in mind with these?

Reply:
I agree both of these hooks are similar yet there are distinctions in what they do:
1. Get data from different queries
2. Map the GraphQL data (`ColonyExtension`) to a custom CDapp entity (`Installed/InstallableExtensionData`) - I don't know a better word for it, "object mapping" sounds like it describes the process best - see https://automapper.org/ or even the `mapPayload` function a few comments above, it has nothing to do with `Array.map()` too. Open to suggestions though
3. In case of `useExtensionData`, return a single extension data, otherwise perform some filtering to split the data into installed and installable sets before returning

I think it's only step 2) that is common for both, so I extracted that into the mapping utils. Hopefully this makes it clearer to you but let me know if you think this can be improved further

Reply:
Thanks for explaining, all good